assets/app/scripts/filters/resources.js

```diff
@@ -845,4 +845,38 @@ angular.module('openshiftConsole')
 
       return portDisplayValue(servicePort.port, servicePort.targetPort, servicePort.protocol);
     };
+  })
+  .filter('routeCondition', function() {
+    return function(route, type) {
+      if (!route.status.ingress) {
+        return null;
+      }
+      var found = null;
+      angular.forEach(route.status.ingress, function(ingress) {
+        angular.forEach(ingress.conditions, function(condition) {
+          if (condition.type === type) {
+            found = condition;
+          }
+        });
+      });
+      return found;
+    };
+  })
+  .filter('routeHost', function() {
+    return function (route) {
+      if (!route.status.ingress || route.status.ingress.length === 0) {
```

Comment:
It will be null

On Feb 15, 2016, at 8:13 PM, Jessica Forrester <notifications@github.com>
wrote:

In assets/app/scripts/filters/resources.js
<https://github.com/openshift/origin/pull/7325#discussion_r52958872>:

> +        return null;
> +      }
> +      var found = null;
> +      angular.forEach(route.status.ingress, function(ingress) {
> +        angular.forEach(ingress.conditions, function(condition) {
> +          if (condition.type === type) {
> +            found = condition;
> +          }
> +        });
> +      });
> +      return found;
> +    };
> +  })
> +  .filter('routeHost', function() {
> +    return function (route) {
> +      if (!route.status.ingress || route.status.ingress.length === 0) {

@smarterclayton <https://github.com/smarterclayton> is route.status.ingress
null, or will it just be an empty array, or could it be either

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/7325/files#r52958872>.